jax/_src/random.py

```diff
@@ -1595,3 +1595,50 @@ def threefry_2x32(keypair, count):
   warnings.warn('jax.random.threefry_2x32 has moved to jax.prng.threefry_2x32 '
                 'and will be removed from `random` module.', FutureWarning)
   return prng.threefry_2x32(keypair, count)
+
+def orthogonal(
+  key: KeyArray,
+  n: int,
+  shape: Tuple[int, ...] = (),
+  dtype: DTypeLikeFloat = dtypes.float_
+) -> jnp.ndarray:
+  """Sample uniformly from the orthogonal group O(n).
+
+  Args:
+    key: a PRNG key used as the random key.
+    n: an integer indicating the resulting dimension.
+    shape: optional, the batch dimensions of the result. Default ().
+    dtype: optional, a float dtype for the returned values (default float64 if
+      jax_enable_x64 is true, otherwise float32).
+
+  Returns:
+    A random array with the specified shape and dtype.
+  """
+  z = normal(key, shape + (n, n), dtype)
```

Comment:
Since `check_shape` lets through arbitrary sequences, it's a bit safer to replace `shape + (n, n)` with `(*shape, n, n)`